### PR TITLE
Correctly generate opaque types

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Based on <https://doc.rust-lang.org/nomicon/ffi.html#representing-opaque-structs>.

You can see the result [here](https://github.com/gtk-rs/gtk3-rs/pull/670).

### CODE REVIEW

File: src/codegen/sys/lib_.rs [hunk 1]
```diff
@@ -389,7 +397,8 @@ fn generate_interfaces_structs(
         if !env.type_status_sys(&full_name).need_generate() {
             continue;
         }
-        writeln!(w, "#[repr(C)]\npub struct {}(c_void);\n", interface.c_type)?;
+        writeln!(w, "#[repr(C)]")?;
```
Comment (sdroege) [inline src/codegen/sys/lib_.rs:400]: Why not move the `#[repr(C)]` into the `generate_opaque_type` function?

File: src/codegen/sys/lib_.rs [hunk 2]
```diff
@@ -489,7 +497,7 @@ fn generate_from_fields(
         // It would be nice to represent those using extern types
         // from RFC 1861, once they are available in stable Rust.
         // https://github.com/rust-lang/rust/issues/43467
-        writeln!(w, "pub struct {name}(c_void);", name = &fields.name)?;
```
Comment (sdroege) [inline src/codegen/sys/lib_.rs:492]: Do we still use `c_void` for something? If not we can remove it from the imports

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise

File: src/codegen/sys/lib_.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply src/codegen/sys/lib_.rs]: I did, but that creates quite the diff haha.

File: src/codegen/sys/lib_.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply src/codegen/sys/lib_.rs:492]: I don't think we do. Let's try removing it. :)

Comment (GuillaumeGomez) [reply src/codegen/sys/lib_.rs:492]:
```
error[E0412]: cannot find type `c_void` in this scope
  --> gdk/sys/src/lib.rs:35:22
   |
35 | pub type GdkXEvent = c_void;
   |                      ^^^^^^ not found in this scope
```

So not yet!

**Comment by GuillaumeGomez** [comment]
Are you ok with the change of position of `repr(C)`?

File: src/codegen/sys/lib_.rs (hunk 1 above)
Comment (sdroege) [reply src/codegen/sys/lib_.rs]: Why?

**Review by sdroege — APPROVED** [review]

File: src/codegen/sys/lib_.rs (hunk 1 above)
Comment (sdroege) [reply src/codegen/sys/lib_.rs]: I see :man_shrugging:

**Pull request merged**